test/unit-test/sigv4_utest.c

```diff
@@ -64,11 +193,191 @@ void formatAndVerifyInputDate( const char * pInputDate,
     tearDown();
 }
 
+/*==================== OpenSSL Based implementation of Crypto Interface ===================== */
+
+static int32_t valid_sha256_init( void * pHashContext )
+{
+    if( SHA256_Init( ( SHA256_CTX * ) pHashContext ) == 1 )
+    {
+        return 0;
+    }
+
+    return -1;
+}
+
+static int32_t valid_sha256_update( void * pHashContext,
+                                    const char * pInput,
+                                    size_t inputLen )
+{
+    if( SHA256_Update( ( SHA256_CTX * ) pHashContext, pInput, inputLen ) )
+    {
+        return 0;
+    }
+
+    return -1;
+}
+
+static int32_t valid_sha256_final( void * pHashContext,
+                                   char * pOutput,
+                                   size_t outputLen )
+{
+    if( SHA256_Final( ( uint8_t * ) pOutput, ( SHA256_CTX * ) pHashContext ) )
+    {
+        return 0;
+    }
+
+    return -1;
+}
+
+/*==================== Echo Implementation of Crypto Interface ===================== */
+
+static hashEchoBuffer[ SIGV4_HASH_MAX_BLOCK_LENGTH ];
```

Comment:
No type? It doesn't look like this is meant to be an `int`, which is the default
```suggestion
static uint8_t hashEchoBuffer[ SIGV4_HASH_MAX_BLOCK_LENGTH ];
```